geomesa-raster/src/test/scala/org/locationtech/geomesa/raster/data/QueryAndMosaicTest.scala

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright 2015 Commonwealth Computer Research, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.locationtech.geomesa.raster.data
+
+import java.awt.image.BufferedImage
+
+import org.junit.runner.RunWith
+import org.locationtech.geomesa.raster.RasterTestsUtils._
+import org.locationtech.geomesa.raster.util.RasterUtils
+import org.locationtech.geomesa.utils.geohash.BoundingBox
+import org.specs2.mutable.Specification
+import org.specs2.runner.JUnitRunner
+
+@RunWith(classOf[JUnitRunner])
+class QueryAndMosaicTest extends Specification {
+  sequential
+
+  var testIteration = 0
+
+  val bboxNorthOf    = BoundingBox(-77.1152343750, -77.104248046875, 43.01220703125, 43.023193359375)
+  val bboxEastOf     = BoundingBox(-77.104248046875, -77.09326171875, 43.001220703125, 43.0122070313125)
+  val bboxOfInterest = BoundingBox(-77.1152343750, -77.104248046875, 43.001220703125, 43.0122070313125)
+  val bboxSouthOf    = BoundingBox(-77.1152343750, -77.104248046875, 42.9902343750, 43.001220703125)
+  val bboxWestOf     = BoundingBox(-77.126220703125, -77.1152343750, 43.001220703125, 43.0122070313125)
+
+  val bboxSouthEastOf  = BoundingBox(-77.104248046875, -77.09326171875, 42.9902343750, 43.001220703125)
+  val bboxSouthWestOf  = BoundingBox(-77.126220703125, -77.1152343750, 42.9902343750, 43.001220703125)
+  val bboxNorthWestOf  = BoundingBox(-77.126220703125, -77.1152343750, 43.01220703125, 43.023193359375)
+  val bboxNorthEastOf  = BoundingBox(-77.104248046875, -77.09326171875, 43.01220703125, 43.023193359375)
+
+  val north:  List[Raster] = (1 to 3).map(i => generateRaster(bboxNorthOf, redHerring, s"$i")).toList
+  val center: List[Raster] = (1 to 3).map(i => generateRaster(bboxOfInterest, testRasterIntVSplit, s"$i")).toList
+  val south:  List[Raster] = (1 to 3).map(i => generateRaster(bboxSouthOf, redHerring, s"$i")).toList
```

Comment:
The example test case we were provided tested three vertical chunks.  Have you tested 3 horizontal ones?  Since you've got the infrastructure, I'd suggest adding those tests.